nixos/modules/services/network-filesystems/webdav.nix

```
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.webdav;
+in
+{
+  options = {
+    services.webdav = {
+      enable = mkEnableOption "WebDAV server";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.webdav;
+        description = "The WebDAV package to use.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "webdav";
+        description = "User account under which WebDAV runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "webdav";
+        description = "Group under which WebDAV runs.";
+      };
+
+      settings = mkOption {
+        type = types.anything;
+        default = { };
+        description = "Attrset that is converted and passed as config file.";
+        example = literalExpression ''
+          {
+              address = "0.0.0.0";
+              port = 8080;
+              scope = "/var/lib/webdav";
+              modify = true;
+              auth = true;
+              users = [
+                {
+                  username = "username";
+                  password = "password";
```

```suggestion
                  username: "{env}ENV_USERNAME"
                  password: "{env}ENV_PASSWORD"
```

Please don't promote storing credentials in the nix store. This software offers the ability to keep credentials outside of configuration and we should promote using the software in this way!